src/client/chunk_closure.cpp

```diff
@@ -414,15 +414,15 @@ void ClientClosure::OnRetry() {
         return;
     }
 
-    if (!reqDone_->IsSuspendRPC() && reqDone_->GetRetriedTimes() >=
-        failReqOpt_.chunkserverMaxRetryTimesBeforeConsiderSuspend) {
-        reqDone_->SetSuspendRPCFlag();
-        MetricHelper::IncremIOSuspendNum(fileMetric_);
-        LOG(ERROR) << "IO Retried "
-                    << failReqOpt_.chunkserverMaxRetryTimesBeforeConsiderSuspend
-                    << " times, set suspend flag! " << *reqCtx_
-                    << ", IO id = " << reqDone_->GetIOTracker()->GetID()
-                    << ", request id = " << reqCtx_->id_;
+    if (CURVE_UNLIKELY(!reqDone_->IsSlowRequest() &&
+                       (TimeUtility::GetTimeofDayMs() - reqDone_->CreatedMS() >
+                        failReqOpt_.chunkserverSlowRequestThresholdMS))) {
+        reqDone_->MarkAsSlowRequest();
+        MetricHelper::IncremSlowRequestNum(fileMetric_);
+        LOG(ERROR) << "Slow request, " << *reqCtx_
```

Comment:
Slow requests will continue to be retried, is it ok to log warnning here?

Reply:
Normally, the IO will complete and return quickly. In an abnormal situation, although IO will keep retrying until processing is complete, we are not aware of this situation. So, a method is needed that can determine that the IO did not return within the expected time. 

Now the threshold is 45 seconds, which should be less likely to be triggered by mistake